### PR TITLE
support llama4

### PR DESCRIPTION
**Pull request by Jintao-Huang**

### CODE REVIEW

**Review by tastelikefeet — APPROVED** [review]

**Review by hjh0119 — APPROVED** [review]

**Comment by Jintao-Huang** [comment]
SFT shell:
```
CUDA_VISIBLE_DEVICES=0,1,2,3 \
swift sft \
    --model LLM-Research/Llama-4-Scout-17B-16E-Instruct \
    --dataset 'AI-ModelScope/LaTeX_OCR:human_handwrite#2000' \
    --train_type lora \
    --torch_dtype bfloat16 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-4 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --freeze_vit true \
    --gradient_accumulation_steps 16 \
    --gradient_checkpointing false \
    --eval_steps 50 \
    --save_steps 50 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --max_length 2048 \
    --output_dir output \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4
```

GPU memory:
截屏2025-04-07 10 06 42

**Comment by Jintao-Huang** [comment]
loss: 
截屏2025-04-07 10 15 54

**Comment by init27** [comment]
@Jintao-Huang Thank you for making this! Would you be interested in adding an example to `llama-cookbook`? Would love to show this FT recipe

**Comment by Jintao-Huang** [comment]
> @Jintao-Huang Thank you for making this! Would you be interested in adding an example to `llama-cookbook`? Would love to show this FT recipe

Thank you for your invitation! We will submit the relevant PR as soon as possible.❤️
